libbeat/common/event.go

```diff
@@ -170,8 +170,29 @@ func normalizeValue(value interface{}, keys ...string) (interface{}, []error) {
 	case bool, []bool:
 	case int, int8, int16, int32, int64:
 	case []int, []int8, []int16, []int32, []int64:
-	case uint, uint8, uint16, uint32, uint64:
-	case []uint, []uint8, []uint16, []uint32, []uint64:
+	case uint, uint8, uint16, uint32:
+	case uint64:
+		return value.(uint64) &^ (1 << 63), nil
```

Comment:
You can replace `value.(uint64)` with `v`. The switch statement gives you typed value — `v`.

Reply:
Umm, there are two different type switches, one for the time types that does the type conversion and this one for the rest that doesn't do it. Is this some kind of optimization? could both switches be merged?

Reply:
It's an optimization. The nested switch falls-back to using reflection to be able to work with types like `type myCustomInt int64`.

Reply:
@andrewkroh this switch-statement does not define the `v` variable. It's the switch statement before the current one. I didn't want to change too much, so I went with `value.(int64)`.

Reply:
My bad. I was looking at the diff on my phone last night 😄 .